### PR TITLE
tests: logging: cleanup

### PR DESCRIPTION
**Pull request by finikorg**
Simple logging tests cleanup

### CODE REVIEW

**Review by cfriedt — DISMISSED** [review]

File: tests/subsys/logging/log_api/src/main.c [hunk 1]
```diff
@@ -430,73 +430,6 @@ ZTEST(test_log_api, test_log_overflow)
...
- * frontend validate number of arguments and values.
- */
-#define MOCK_LOG_FRONT_BACKEND_RECORD(str) do { \
-	mock_log_backend_record(&backend1, LOG_CURRENT_MODULE_ID(), \
-				Z_LOG_LOCAL_DOMAIN_ID, LOG_LEVEL_INF, \
-				exp_timestamp++, str); \
-	mock_log_frontend_record(LOG_CURRENT_MODULE_ID(), LOG_LEVEL_INF, str); \
-} while (0)
-
-ZTEST(test_log_api, test_log_arguments)
```
Comment (dcpleung) [inline tests/subsys/logging/log_api/src/main.c:446]: This shouldn't be dead code, right? IIRC, this runs as part of the test.

Comment (cfriedt) [reply tests/subsys/logging/log_api/src/main.c:446]: Hmm... yeah, I missed that actually.

Comment (finikorg) [reply tests/subsys/logging/log_api/src/main.c:446]: what is the first operation of this test? `return;`

Comment (dcpleung) [reply tests/subsys/logging/log_api/src/main.c:446]: @nordic-krch do you remember why this test returns immediately?

**Review by nordic-krch — APPROVED** [review]

File: tests/subsys/logging/log_api/src/main.c (hunk 1 above)
Comment (nordic-krch) [reply tests/subsys/logging/log_api/src/main.c:446]: It was accidentally added by https://github.com/zephyrproject-rtos/zephyr/commit/3dc12753ff05dad8612c33d3be803c7078065e30#diff-1c377e080a9a6230d6bd46983219fcf6dde1e5b7b58ef36061fd9a2cff7b0ff0 so rather return should be removed but test case ketp.

**Review by nordic-krch — CHANGES_REQUESTED** [review]

File: tests/subsys/logging/log_api/src/main.c (hunk 1 above)
Comment (finikorg) [reply tests/subsys/logging/log_api/src/main.c:446]: I believe `return` was added in purpose, otherwise tests are failing.

Comment (nordic-krch) [reply tests/subsys/logging/log_api/src/main.c:446]: On which platform? I tried on `qemu_x86` and it passes.

Comment (finikorg) [reply tests/subsys/logging/log_api/src/main.c:446]:
```
$ twister -c -T zephyr/tests/subsys/logging/log_api/ -p qemu_x86_64
```

Comment (nordic-krch) [reply tests/subsys/logging/log_api/src/main.c:446]: Test case fixed in #53538.

Comment (finikorg) [reply tests/subsys/logging/log_api/src/main.c]: I have removed that commit.

File: tests/subsys/logging/log_cache/src/main.c [hunk 2]
```diff
@@ -139,15 +139,15 @@ ZTEST(test_log_cache, test_log_cache_basic)
 
 	/* id0 is evicted since it is the oldest one. */
 	cache_get(&cache, id0.raw, &buf, false, __LINE__);
-	buf_check(buf, 1);
+	zassert_true(buf_check(buf, 2), "Buffer check failed");
```
Comment (nordic-krch) [inline tests/subsys/logging/log_cache/src/main.c:142]: is it intentional to change  `buf_check(buf,1)` to `buf_check(buf, 2)`?

Comment (finikorg) [reply tests/subsys/logging/log_cache/src/main.c:142]: yes, there is special commit for that

**Review by nordic-krch — DISMISSED** [review]

**Review by dcpleung — DISMISSED** [review]

**Comment by finikorg** [comment]
Rebased because of failing CI.

**Review by dcpleung — APPROVED** [review]

**Review by cfriedt — APPROVED** [review]

**Pull request merged**